tempest/api/workloadmgr/workload_reassign/test_workload_reassign_same_project_same_user.py

```diff
@@ -0,0 +1,170 @@
+import os
+import sys
+
+from oslo_log import log as logging
+
+from tempest import config
+from tempest import reporting
+from tempest import tvaultconf
+from tempest.api.workloadmgr import base
+from tempest.lib import decorators
+
+sys.path.append(os.getcwd())
+
+LOG = logging.getLogger(__name__)
+CONF = config.CONF
+
+
+class WorkloadTest(base.BaseWorkloadmgrTest):
+
+    credentials = ['primary']
+
+    @classmethod
+    def setup_clients(cls):
+        super(WorkloadTest, cls).setup_clients()
+        reporting.add_test_script(str(__name__))
+
+    def create_snapshot(self, workload_id, is_full=True):
+        if is_full:
+            substitution = 'Full'
+        else:
+            substitution = 'Incremental'
+
+        snapshot_id, command_execution, snapshot_execution = self.workload_snapshot_cli(
+            workload_id, is_full=is_full)
+        if command_execution == 'pass':
+            reporting.add_test_step("{} snapshot command execution".format(
+                substitution), tvaultconf.PASS)
+            LOG.debug("Command executed correctly for full snapshot")
+        else:
+            reporting.add_test_step("{} snapshot command execution".format(
+                substitution), tvaultconf.FAIL)
+            raise Exception(
+                "Command did not execute correctly for full snapshot")
+
+        if snapshot_execution == 'pass':
+            reporting.add_test_step("{} snapshot".format(
+                substitution), tvaultconf.PASS)
+        else:
+            reporting.add_test_step("{} snapshot".format(
+                substitution), tvaultconf.FAIL)
+            raise Exception("Full snapshot failed")
+
+        return (snapshot_id)
+
+
+    @decorators.idempotent_id('9fe07175-912e-49a5-a629-5f52eeada4c9')
+    @decorators.attr(type='workloadmgr_api')
+    def test_workload_reassign_same_project_same_user(self):
+        try:
+
+            tests = [['tempest.api.workloadmgr.workload_reassignment.create_workload_and_snapshot',
+                      0],
+                     ['tempest.api.workloadmgr.workload_reassignment.assign_tenant1_and_user1',
```

Comment:
Assigning tenant and user can also be added as test step but not a test case.

Reply:
this I kept as separate for keeping workload reassignment with same user and same tenant in 2 stages for verification.